packages/gatsby-transformer-remark/src/extend-node-type.js

```diff
@@ -165,35 +165,21 @@ module.exports = (
       }
     }
 
-    async function getMarkdownAST(markdownNode) {
-      if (process.env.NODE_ENV !== `production` || !fileNodes) {
-        fileNodes = getNodesByType(`File`)
+    // Parse a markdown string and its AST representation,
+    // applying the remark plugins if necesserary
+    async function parseString(string, markdownNode) {
+      // compiler to inject in the remark plugins
+      // so that they can use our parser/generator
+      // with all the options and plugins from the user
+      const compiler = {
+        parseString: string => parseString(string, markdownNode),
+        generateHTML: ast =>
+          hastToHTML(ast, {
```

Comment:
I think this still has to be converted to HAST first (as we basically get remark AST here):

```suggestion
          hastToHTML(toHAST(ast), {
```

Reply:
you're right, and there are actually a couple of options we pass to that function so I extracted it to use it everywhere